hummingbot/connector/exchange/bittrex/bittrex_exchange.py

```diff
@@ -0,0 +1,403 @@
+
+    async def _place_order(self,
+                           order_id: str,
+                           trading_pair: str,
+                           amount: Decimal,
+                           trade_type: TradeType,
+                           order_type: OrderType,
+                           price: Decimal,
+                           **kwargs
+                           ) -> Tuple[str, float]:
+
+        path_url = CONSTANTS.ORDER_CREATION_URL
+        body = {
+            "marketSymbol": await self.exchange_symbol_associated_to_pair(trading_pair=trading_pair),
+            "direction": trade_type.name,
+            "type": "LIMIT",
+            "quantity": f"{amount:f}",
+            "limit": f"{price:f}",
+        }
+        if order_type is OrderType.LIMIT:
+            body.update({
+                "timeInForce": "GOOD_TIL_CANCELLED"
+            })
+        elif order_type is OrderType.LIMIT_MAKER:
+            body.update({
+                "timeInForce": "POST_ONLY_GOOD_TIL_CANCELLED"
+            })
+        body.update({
+            "clientOrderId": order_id
+        })
+        order_result = await self._api_post(
+            path_url=path_url,
+            data=body,
+            is_auth_required=True)
+        o_id = order_result["id"]
+        transact_time = _get_timestamp(order_result["createdAt"])
+        return o_id, transact_time
+
+    async def _place_cancel(self, order_id: str, tracked_order: InFlightOrder):
+        exchange_id = await tracked_order.get_exchange_order_id()
+        api_params = {
+            "orderId": exchange_id
+        }
+        cancel_result = await self._api_delete(
+            path_url=CONSTANTS.ORDER_DETAIL_URL.format(exchange_id),
+            params=api_params,
+            is_auth_required=True,
+            limit_id=CONSTANTS.ORDER_DETAIL_LIMIT_ID)
+        if cancel_result["status"] == "CLOSED":
+            return True
+        return False
+
+    async def _update_balances(self):
+        local_asset_names = set(self._account_balances.keys())
+        remote_asset_names = set()
+        account_balances = await self._api_get(path_url=CONSTANTS.BALANCES_URL, is_auth_required=True)
+        for balance_entry in account_balances:
+            asset_name = balance_entry["currencySymbol"]
+            available_balance = Decimal(balance_entry["available"])
+            total_balance = Decimal(balance_entry["total"])
+            self._account_available_balances[asset_name] = available_balance
+            self._account_balances[asset_name] = total_balance
+            remote_asset_names.add(asset_name)
+        asset_names_to_remove = local_asset_names.difference(remote_asset_names)
+        for asset_name in asset_names_to_remove:
+            del self._account_available_balances[asset_name]
+            del self._account_balances[asset_name]
+
+    async def _format_trading_rules(self, markets: List) -> List[TradingRule]:
+        retval = []
+        for market in markets:
```

Comment:
We should apply to the markets we will process for trading rules the same filter/selection criteria we apply when initializing the trading pairs (in the method `def _initialize_trading_pair_symbols_from_exchange_info(self, exchange_info: List):`)